### ISSUE TITLE
Can not search on ttPlayer

### ISSUE THREAD

**Issue by GoogleCodeExporter** [memeco/osd-lyrics#222]
```
What steps will reproduce the problem?
请列出导致问题的操作
1.不使用代理、使用代理，均无法搜索到歌词。
2.
3.

What's the output in terminal (run with command `osd-lyrics -d debug')?
请给出在终端下运行的输出（使用“osd-lyrics -d
debug”来运行）
*Debug*: in function ol_player_exaile03_get: ol_player_exaile03.c[412]
ol_player_exaile03_get
*Debug*: in function ol_player_listen_get: ol_player_listen.c[360]
ol_player_listen_get
*Debug*: in function ol_player_gmusicbrowser_get: ol_player_gmusicbrowser.c[375]
ol_player_gmusicbrowser_get
*Debug*: in function ol_config_class_init: ol_config.c[152]
id of changed signal is: 146
*Debug*: in function ol_config_get_path: ol_config.c[363]
config path: /home/prcxjb/.config/osdlyrics/osdlyrics.conf
*Debug*: in function ol_config_init: ol_config.c[100]
active-lrc-color
*Debug*: in function ol_config_init: ol_config.c[100]
inactive-lrc-color
*Debug*: in function ol_config_init: ol_config.c[100]
lrc-path
*Debug*: in function ol_config_init: ol_config.c[100]
lrc-filename
*Debug*: in function ol_config_get_string: ol_config.c[336]
[General]display-mode:OSD
*Debug*: in function ol_osd_render_update_font: ol_osd_render.c[240]
serif 30
*Debug*: in function ol_osd_window_set_bg: ol_osd_window.c[1874]
bg: 0x8eb8ea8
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]width
*Debug*: in function ol_config_get_int: ol_config.c[310]
[OSD]width:1024
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]osd-window-mode
*Debug*: in function ol_config_get_string: ol_config.c[336]
[OSD]osd-window-mode:dock
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]visible
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]visible:1
*Debug*: in function ol_image_button_size_request: ol_image_button.c[67]
request: 22 x 22
*Debug*: in function ol_image_button_size_request: ol_image_button.c[67]
request: 22 x 22
*Debug*: in function ol_image_button_size_request: ol_image_button.c[67]
request: 22 x 22
*Debug*: in function ol_image_button_size_request: ol_image_button.c[67]
request: 22 x 22
*Debug*: in function ol_image_button_size_request: ol_image_button.c[67]
request: 22 x 22
*Debug*: in function ol_osd_window_update_child_allocation: ol_osd_window.c[919]
set child allocation
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (457, 59) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (479, 59) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (501, 59) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (523, 59) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (545, 59) 22 x 22
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]locked
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]locked:1
*Debug*: in function config_change_handler: ol_osd_module.c[217]
  locked: 1
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]locked:1
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]line-count
*Debug*: in function ol_config_get_int: ol_config.c[310]
[OSD]line-count:1
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]font-family
*Debug*: in function ol_config_get_string: ol_config.c[336]
[OSD]font-family:微软雅黑-Tahoma
*Debug*: in function ol_osd_render_update_font: ol_osd_render.c[240]
微软雅黑-Tahoma 30
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]font-size
*Debug*: in function ol_config_get_double: ol_config.c[323]
[OSD]font-size:30.000000
*Debug*: in function ol_osd_render_update_font: ol_osd_render.c[240]
微软雅黑-Tahoma 30
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]x
*Debug*: in function ol_config_get_int: ol_config.c[310]
[OSD]y:0
*Debug*: in function ol_config_get_int: ol_config.c[310]
[OSD]x:0
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]lrc-align-0
*Debug*: in function ol_config_get_double: ol_config.c[323]
[OSD]lrc-align-0:0.000000
*Debug*: in function ol_osd_window_set_line_alignment: ol_osd_window.c[1238]
ol_osd_window_set_line_alignment:0-0.000000
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]lrc-align-1
*Debug*: in function ol_config_get_double: ol_config.c[323]
[OSD]lrc-align-1:1.000000
*Debug*: in function ol_osd_window_set_line_alignment: ol_osd_window.c[1238]
ol_osd_window_set_line_alignment:1-1.000000
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]active-lrc-color
*Debug*: in function config_change_handler: ol_osd_module.c[258]
len = 3
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]inactive-lrc-color
*Debug*: in function config_change_handler: ol_osd_module.c[272]
len = 3
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]translucent-on-mouse-over
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]translucent-on-mouse-over:0
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]outline-width
*Debug*: in function ol_config_get_int: ol_config.c[310]
[OSD]outline-width:0
*Debug*: in function ol_config_changed: ol_menu.c[154]
group:OSD  name:locked
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]locked:1
*Debug*: in function config_change_handler: ol_osd_module.c[205]
config_change_handler:[OSD]locked
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]locked:1
*Debug*: in function config_change_handler: ol_osd_module.c[217]
  locked: 1
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]locked:1
*Debug*: in function ol_config_changed: ol_menu.c[154]
group:OSD  name:visible
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[OSD]visible:1
*Debug*: in function ol_config_changed: ol_menu.c[154]
group:General  name:display-mode
*Debug*: in function ol_config_get_string: ol_config.c[336]
[General]display-mode:OSD
*Debug*: in function ol_lrc_fetch_add_engine: ol_lrc_fetch.c[95]
engine:ttPlayer
*Debug*: in function ol_osd_window_update_child_allocation: ol_osd_window.c[919]
set child allocation
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (457, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (479, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (501, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (523, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (545, 55) 22 x 22
*Debug*: in function ol_player_get_active_player: ol_player.c[128]
controller count:21
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Amarok 2
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.kde.amarok': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Banshee
*Debug*: in function ol_player_banshee_init_dbus: ol_player_banshee.c[192]
get proxy failed: Could not get owner of name 'org.bansheeproject.Banshee': no 
such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Exaile 0.2
*Debug*: in function ol_player_exaile02_init_dbus: ol_player_exaile02.c[246]
get proxy failed: Could not get owner of name 'org.exaile.DBusInterface': no 
such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Exaile 0.3
*Debug*: in function ol_player_exaile03_init_dbus: ol_player_exaile03.c[314]
get proxy failed: Could not get owner of name 'org.exaile.Exaile': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Audacious
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.mpris.audacious': no such 
name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Songbird
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.mpris.songbird': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Clementine
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.mpris.clementine': no such 
name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Listen Music Player
*Debug*: in function ol_player_listen_init_dbus: ol_player_listen.c[264]
get proxy failed: Could not get owner of name 'org.gnome.Listen': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying guayadeque 0.2.5
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.mpris.guayadeque': no such 
name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Deciber
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.mpris.dap': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying Gmusicbrowser
*Debug*: in function ol_player_gmusicbrowser_init_dbus: 
ol_player_gmusicbrowser.c[279]
get proxy failed: Could not get owner of name 'org.gmusicbrowser': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying vlc
*Debug*: in function ol_player_mpris_init_dbus: ol_player_mpris.c[255]
get proxy failed: Could not get owner of name 'org.mpris.vlc': no such name
*Debug*: in function ol_player_get_active_player: ol_player.c[132]
trying QMMP
*Debug*: in function ol_osd_window_update_child_allocation: ol_osd_window.c[919]
set child allocation
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (468, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (490, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (512, 55) 22 x 22
*Debug*: in function ol_image_button_size_allocate: ol_image_button.c[77]
allocation: (534, 55) 22 x 22
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function _find_by_uri: ol_lrclib.c[133]
SELECT lrcpath FROM lyrics WHERE uri = 'cue:///media/study/音乐/eagles/Eagles 
- (1976) Their Greatest Hits 1971-1975/Eagles - (1976) [1976 Asylum 105-2 
(253-017)] Their Greatest Hits 1971-1975.cue#4'
*Debug*: in function ol_lrclib_find: ol_lrclib.c[366]
found = 0
*Debug*: in function _find_by_info: ol_lrclib.c[177]
SELECT lrcpath FROM lyrics WHERE title = 'Already Gone' AND artist = 'Eagles' 
AND album = '' AND uri is NULL
*Debug*: in function internal_for_each: ol_lyric_manage.c[47]
  uri: cue:///media/study/音乐/eagles/Eagles - (1976) Their Greatest Hits 1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 1971-1975.cue#4
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:~/.lyrics, name:%p-%t
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[326]
  /home/prcxjb/.lyrics/Eagles-Already Gone.lrc
*Debug*: in function internal_check_lyric_file: ol_lyric_manage.c[62]
  filename:/home/prcxjb/.lyrics/Eagles-Already Gone.lrc
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:~/.lyrics, name:%t-%p
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[326]
  /home/prcxjb/.lyrics/Already Gone-Eagles.lrc
*Debug*: in function internal_check_lyric_file: ol_lyric_manage.c[62]
  filename:/home/prcxjb/.lyrics/Already Gone-Eagles.lrc
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:~/.lyrics, name:%f
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[326]
  /home/prcxjb/.lyrics/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 1971-1975.lrc
*Debug*: in function internal_check_lyric_file: ol_lyric_manage.c[62]
  filename:/home/prcxjb/.lyrics/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 1971-1975.lrc
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:~/.lyrics, name:%t
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[326]
  /home/prcxjb/.lyrics/Already Gone.lrc
*Debug*: in function internal_check_lyric_file: ol_lyric_manage.c[62]
  filename:/home/prcxjb/.lyrics/Already Gone.lrc
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:%, name:%p-%t
*Error*: in function ol_uri_get_path: ol_path_pattern.c[250]
Cannot get pathname from uri cue:///media/study/音乐/eagles/Eagles - (1976) 
Their Greatest Hits 1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] 
Their Greatest Hits 1971-1975.cue#4: 
URI“cue:///media/study/音乐/eagles/Eagles - (1976) Their Greatest Hits 
1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 
1971-1975.cue#4”不是“file”格式的绝对 URI
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:%, name:%t-%p
*Error*: in function ol_uri_get_path: ol_path_pattern.c[250]
Cannot get pathname from uri cue:///media/study/音乐/eagles/Eagles - (1976) 
Their Greatest Hits 1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] 
Their Greatest Hits 1971-1975.cue#4: 
URI“cue:///media/study/音乐/eagles/Eagles - (1976) Their Greatest Hits 
1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 
1971-1975.cue#4”不是“file”格式的绝对 URI
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:%, name:%f
*Error*: in function ol_uri_get_path: ol_path_pattern.c[250]
Cannot get pathname from uri cue:///media/study/音乐/eagles/Eagles - (1976) 
Their Greatest Hits 1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] 
Their Greatest Hits 1971-1975.cue#4: 
URI“cue:///media/study/音乐/eagles/Eagles - (1976) Their Greatest Hits 
1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 
1971-1975.cue#4”不是“file”格式的绝对 URI
*Debug*: in function ol_path_pattern_for_each: ol_path_pattern.c[319]
  path:%, name:%t
*Error*: in function ol_uri_get_path: ol_path_pattern.c[250]
Cannot get pathname from uri cue:///media/study/音乐/eagles/Eagles - (1976) 
Their Greatest Hits 1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] 
Their Greatest Hits 1971-1975.cue#4: 
URI“cue:///media/study/音乐/eagles/Eagles - (1976) Their Greatest Hits 
1971-1975/Eagles - (1976) [1976 Asylum 105-2 (253-017)] Their Greatest Hits 
1971-1975.cue#4”不是“file”格式的绝对 URI
*Debug*: in function _check_lyric_file: ol_main.c[232]
filename;(null)
*Debug*: in function ol_config_get_string: ol_config.c[336]
[Download]download-engine:ttPlayer
*Debug*: in function ol_app_download_lyric: ol_main.c[171]
Download engine: ttPlayer
*Debug*: in function ol_lrc_fetch_get_engine: ol_lrc_fetch.c[79]
[0]:ttPlayer
*Debug*: in function ol_lrc_fetch_begin_search: ol_lrc_fetch_module.c[200]
  title: Already Gone
  artist: Eagles
  album: 
*Debug*: in function ol_osd_module_set_message: ol_osd_module.c[560]
  message:正在搜索歌词
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:正在搜索歌词
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function ol_config_get_bool: ol_config.c[297]
[General]notify-music:1
*Debug*: in function _get_notify: ol_notify.c[49]
summary: Already Gone
body: Eagles
<i></i>
icon: /usr/share/icons/hicolor/48x48/apps/qmmp.png
*Debug*: in function _search: ol_lrc_fetch_ttplayer.c[186]
url: 
http://lrcct2.ttplayer.com/dll/lyricsvr.dll?sh?Artist=6500610067006c0065007300&T
itle=61006c007200650061006400790067006f006e006500&Flags=0
*Debug*: in function ol_config_get_string: ol_config.c[336]
[Download]proxy:no
*Error*: in function fetch_into_memory: ol_lrc_fetch_utils.c[329]
failed to perform: [Couldn't resolve host 'lrcct2.ttplayer.com']
*Debug*: in function _search: ol_lrc_fetch_ttplayer.c[190]
Search lyrics failed
*Debug*: in function internal_search_callback: ol_lrc_fetch_module.c[101]
returned data:
0
*Debug*: in function ol_osd_module_set_message: ol_osd_module.c[560]
  message:找不到匹配的歌词
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:找不到匹配的歌词
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)
*Debug*: in function ol_osd_window_set_lyric: ol_osd_window.c[1209]
  lyric:(null)

What's the player and which version you are using?
您使用的播放器是什么？它的版本号是多少？
qmmp，0.4.3。banshee，2.0.1
What's version of OSD Lyrics?
OSD Lyrics的版本是多少？
0.4.0
What's your Linux distro?
您使用哪种发行版？
ubuntu 11.04

```

Original issue reported on code.google.com by `xjb...@gmail.com` on 27 Jun 2011 at 1:42

**Comment by GoogleCodeExporter** [comment]
```
千千的旧域名废弃了，下一版本会修正
```

Original comment by `tigerso...@gmail.com` on 29 Jun 2011 at 3:26
- Changed title: **Can not search on ttPlayer**
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
$ osd-lyrics
*Error*: in function fetch_into_memory: ol_lrc_fetch_utils.c[329]
failed to perform: [Couldn't resolve host 'lrcct2.ttplayer.com']
$ host lrcct2.ttplayer.com               
Host lrcct2.ttplayer.com not found: 3(NXDOMAIN)
$ whois ttplayer.com | grep "Name Server"
Name Server: NS1.DNS.COM.CN
Name Server: NS2.DNS.COM.CN
$ host lrcct2.ttplayer.com ns1.dns.com.cn
Using domain server:
Name: ns1.dns.com.cn
Address: 211.100.44.201#53
Aliases: 

lrcct2.ttplayer.com has address 220.181.5.39

Bottom line, ttplayer.com has got its DNS records all messed up because, for 
some reason, the records are only stored on their Chinese DNS server.

That being said,, adding 220.181.5.39 to your host file should fix the problem.

Under Linux: $ sudo echo -e "220.181.5.39\tlrcct2.ttplayer.com" >> 
/etc/hostslrcct2.ttplayer.com
```

Original comment by `p...@fusi0n.org` on 2 Jul 2011 at 5:36

**Comment by GoogleCodeExporter** [comment]
```
Issue 224 has been merged into this issue.
```

Original comment by `tigerso...@gmail.com` on 2 Jul 2011 at 5:47

**Comment by GoogleCodeExporter** [comment]
```
That last line obviously should have read:

$ sudo echo -e "220.181.5.39\tlrcct2.ttplayer.com" >> /etc/hosts

Apologies for the impromptu cut-and-paste.
```

Original comment by `p...@fusi0n.org` on 2 Jul 2011 at 7:21
